builtin/game/chat.lua

```diff
@@ -1234,33 +1234,67 @@ core.register_chatcommand("kick", {
 	end,
 })
 
+local clearobject_confirms = {}
+
 core.register_chatcommand("clearobjects", {
-	params = S("[full | quick]"),
+	params = S("[full | quick | confirm | cancel]"),
 	description = S("Clear all objects in world"),
 	privs = {server=true},
 	func = function(name, param)
-		local options = {}
-		if param == "" or param == "quick" then
-			options.mode = "quick"
-		elseif param == "full" then
-			options.mode = "full"
-		else
-			return false, S("Invalid usage, see /help clearobjects.")
-		end
+		if param == "" or param == "quick" or param == "full" then
+			local options = {
+				mode = param == "" and "quick" or param
+			}
+			if clearobject_confirms[name] then
+				clearobject_confirms[name][1]:cancel()
+			end
+			clearobject_confirms[name] = {
+				core.after(60, function()
```

Comment:
the need to `:cancel` these seems unwieldly.
personally I would just save the timestamp and compare it on `confirm`, that requires no removal job

Reply:
Done in 2a0236b34